src/plugins/kibana_react/public/page_template/with_solution_navbar.tsx

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { useState } from 'react';
+import classNames from 'classnames';
+import { useIsWithinBreakpoints } from '@elastic/eui';
+import { EuiPageSideBarProps } from '@elastic/eui/src/components/page/page_side_bar';
+import { KibanaPageTemplateSolutionNav } from '../page_template/solution_nav';
+import { KibanaPageTemplateProps } from '../page_template';
+
+export const withSolutionNavbar = (
+  WrappedComponent: React.FunctionComponent<KibanaPageTemplateProps>
+) => {
+  const WithSolutionNavBar = (props: KibanaPageTemplateProps) => {
+    const isMediumBreakpoint = useIsWithinBreakpoints(['m']);
+    const isLargerBreakpoint = useIsWithinBreakpoints(['l', 'xl']);
+    const [isSideNavOpenOnDesktop, setisSideNavOpenOnDesktop] = useState(
+      !JSON.parse(String(localStorage.getItem('solutionNavIsCollapsed')))
+    );
+    const { solutionNav, children, isEmptyState, template } = props;
+    if (!solutionNav) {
+      return <WrappedComponent {...props} />;
+    }
```

Comment:
If this component is specifically called `WithSolutionNav` shouldn't `solutionNav` be required, making this return unnecessary?